src/formatDate.js

```diff
@@ -8,7 +8,85 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [aFrom, bFrom, cFrom, separator] = fromFormat;
+  const fromFormatArray = [aFrom, bFrom, cFrom];
+  const [aTo, bTo, cTo, newSeparator] = toFormat;
+  const toFormatArray = [aTo, bTo, cTo];
+  const dateinNumbers = date.split(separator);
+
+  const dateInfo = getDateFromOld(fromFormatArray, dateinNumbers);
+
+  return setDateOrderNew(toFormatArray, dateInfo).join(newSeparator);
 }
 
+function getDateFromOld(fromArray, dateNumbers) {
+  const dateObj = {};
+
+  for (let i = 0; i < fromArray.length; i++) {
+    switch (fromArray[i]) {
+      case 'DD': {
+        dateObj.day = dateNumbers[i];
+        break;
+      }
+
+      case 'MM': {
+        dateObj.month = dateNumbers[i];
+        break;
+      }
+
+      case 'YYYY': {
+        dateObj.year = dateNumbers[i];
+        break;
+      }
+
+      case 'YY': {
+        if (dateNumbers[i] < 30) {
+          dateObj.year = `20${dateNumbers[i]}`;
+        } else {
+          dateObj.year = `19${dateNumbers[i]}`;
```

Comment:
The logic for determining the century for 'YY' format assumes that any year less than 30 belongs to the 2000s, and any year 30 or greater belongs to the 1900s. This is a common heuristic but may not be universally applicable. Consider making this logic configurable or document this assumption clearly.